crates/testing/tests/tests_cx_hardening/cxh_da_task.rs

```diff
@@ -0,0 +1,564 @@
+            .expect("Failed to initialize HotShot");
+
+    // Prepare empty transactions and compute a commitment for later use
+    let transactions = vec![TestTransaction::new(vec![0])];
+    let encoded_transactions = Arc::from(TestTransaction::encode(&transactions));
+    let (payload_commit, _precompute) = precompute_vid_commitment(
+        &encoded_transactions,
+        handle.hotshot.memberships.quorum_membership.total_nodes(),
+    );
+
+    // Initialize a view generator using the current memberships
+    let mut view_generator = TestViewGenerator::generate(
+        handle.hotshot.memberships.quorum_membership.clone(),
+        handle.hotshot.memberships.da_membership.clone(),
+    );
+
+    // Generate views for the test
+    let view1 = view_generator.next().await.unwrap();
+    let _view2 = view_generator.next().await.unwrap();
+    view_generator.add_transactions(transactions);
+    let view3 = view_generator.next().await.unwrap();
+
+    // Define input events for the test:
+    // 1. Three view changes and an outdated proposal for view 1 when in view 3
+    // 2. A current proposal for view 3
+    let inputs = vec![
+        serial![
+            HotShotEvent::ViewChange(ViewNumber::new(1)),
+            HotShotEvent::ViewChange(ViewNumber::new(2)),
+            HotShotEvent::ViewChange(ViewNumber::new(3)),
+            // Send an outdated proposal (view 1) when we're in view 3
+            HotShotEvent::DaProposalRecv(view1.da_proposal.clone(), view1.leader_public_key),
+        ],
+        serial![
+            // Send a current proposal (view 3)
+            HotShotEvent::DaProposalRecv(view3.da_proposal.clone(), view3.leader_public_key),
+        ],
+    ];
+
+    // Define expectations:
+    // 1. No output for the outdated proposal
+    // 2. Validation and voting actions for the current proposal
+    let expectations = vec![
+        Expectations::from_outputs(vec![]),
+        Expectations::from_outputs(vec![
+            exact(HotShotEvent::DaProposalValidated(
+                view3.da_proposal.clone(),
+                view3.leader_public_key,
+            )),
+            exact(HotShotEvent::DaVoteSend(view3.create_da_vote(
+                DaData {
+                    payload_commit,
+                },
+                &handle,
+            ))),
+        ]),
+    ];
+
+    // Define expectations for external events triggered by the system
+    let external_event_expectations = vec![expect_external_events(vec![ext_event_exact(Event {
+        view_number: view3.view_number,
+        event: EventType::DaProposal {
+            proposal: view3.da_proposal.clone(),
+            sender: view3.leader_public_key,
+        },
+    })])];
+
+    // Create DA task state and script for the test
+    let da_state = DaTaskState::<TestTypes, MemoryImpl>::create_from(&handle).await;
+    let mut da_script = TaskScript {
+        timeout: Duration::from_millis(100),
+        state: da_state,
+        expectations,
+    };
```

Comment:
I don't think we need these external event handlers as the `DaProposal` event is transformed into `DaProposalRecv` in the downstream handlers.

Reply:
Related to the comment, https://github.com/EspressoSystems/HotShot/pull/3568#discussion_r1715774905

Reply:
I think I still feel strongly that we don't necessarily need to process the external events. If the event is transformed successfully into a, for example, `DaProposalRecv` event, then we *know* with certainty that the external event was received, so we don't need to handle this stream in that case. To avoid bespoke changes, I'd prefer to remove it since the checks are redundant under this circumstance.